copycat/api/src/main/java/org/apache/kafka/copycat/sink/SinkTaskContext.java

```diff
@@ -59,6 +63,40 @@ public void offset(Map<TopicPartition, Long> offsets) {
     }
 
     /**
+     * Set the backoff timeout. SinkTasks should use this to indicate that they need to retry certain
+     * operations after the backoff timeout.
+     * @param backoffMs the backoff timeout in milliseconds.
+     */
+    public void backoffMs(long backoffMs) {
+        this.backoffMs = backoffMs;
+    }
+
+    /**
+     * Get the backoff in milliseconds set by SinkTasks. Used by the Copycat framework.
+     * @return the backoff timeout in milliseconds.
+     */
+    public long backoffMs() {
+        return backoffMs;
+    }
+
+    /**
+     * Set the topics that needs to rewind. SinkTask should use this to indicate that some topic
+     * partitions needs to rewind.
+     * @param rewinds The set of topic partitions that needs to rewind.
+     */
+    public void rewinds(Set<TopicPartition> rewinds) {
```

Comment:
Same here -- should probably just be `public void rewind(Set<TopicPartition> partitions)`.

Also, is there a reason we need both `offsets()` and `rewinds()`? Seems like they accomplish basically the same thing, but with slightly different semantics. Why not have just one mechanism for requesting offset changes to the framework, and on each iteration, the `WorkerSinkTask` can handle checking if there are offsets that need to be reset, reset them, and then clear out that list?

The only benefit I see right now is that by having one of them be stateful, you've moved a bit of the work of maintaining the offsets to the framework, but I'm not sure that small benefit to the task developer is worth having nearly duplicate functionality.